### ISSUE TITLE
Uncaught TypeError: Cannot read property 'value' of undefined

### ISSUE THREAD

**Issue by softmarshmallow** [suren-atoyan/monaco-react#237]
with NextJS, next-transpile-modules, having this error, "Loading" Text on screen.

```
Uncaught TypeError: Cannot read property 'value' of undefined
    at C (decorators.ts:12)
    at M (decorators.ts:60)
    at eval (index.js?b7ab:36)
    at Array.forEach (<anonymous>)
    at decorateConstructor (index.js?b7ab:35)
    at Reflect.decorate (index.js?b7ab:10)
    at Me (globalMouseMoveMonitor.ts:41)
    at touch.ts:124
    at Function.r._invokeFactory (loader.js:1118)
    at r.complete (loader.js:1128)
    at r._onModuleComplete (loader.js:1754)
    at r._resolve (loader.js:1714)
    at r.defineModule (loader.js:1357)
    at _ (loader.js:1804)
    at globalMouseMoveMonitor.ts:41
    at fake:1
```

![image](https://user-images.githubusercontent.com/16307013/116762979-1ff5a100-aa57-11eb-8a2e-921d6443f841.png)

Which is throw here (restored via source map)
```

	const result = function memoize(target: any, key: string, descriptor: any) {
		let fnKey: string | null = null;
		let fn: Function | null = null;

		if (typeof descriptor.value === 'function') {
			fnKey = 'value';
			fn = descriptor.value;

			if (fn!.length !== 0) {
				console.warn('Memoize should only be used in functions with zero parameters');
			}
		} else if (typeof descriptor.get === 'function') {
			fnKey = 'get';
			fn = descriptor.get;
		}
```

**Comment by softmarshmallow** [comment]
@suren-atoyan  Can you help me with this?

**Comment by suren-atoyan** [comment]
> @suren-atoyan Can you help me with this?

could you please provide a reproducible code snippet?

**Comment by softmarshmallow** [comment]
@suren-atoyan  This error seems complicated.


https://user-images.githubusercontent.com/16307013/120512286-2e442d80-c406-11eb-8027-ab467ba99319.mov



I'm using exact same layout for two pages.
1. ~/to-react
2. ~/to-flutter

the imports are
to-react.tsx
```
import React, { useState } from "react";
import styled from "@emotion/styled";
import { MainImageRepository } from "@design-sdk/core/assets-repository";
import { ImageRepositories } from "@design-sdk/figma/asset-repository";
import { figmacomp } from "../../components";
import { ReflectSceneNode } from "@design-sdk/core/nodes";
import { DefaultEditorWorkspaceLayout } from "../../layout/default-editor-workspace-layout";
import { LayerHierarchy } from "../../components/editor-hierarchy";
import { PreviewAndRunPanel } from "../../components/preview-and-run";
import { FigmaTargetNodeConfig } from "@design-sdk/core/utils/figma-api-utils";
import {
  WorkspaceContentPanel,
  WorkspaceContentPanelGridLayout,
} from "../../layout/panel";
import { WorkspaceBottomPanelDockLayout } from "../../layout/panel/workspace-bottom-panel-dock-layout";
import { JsonTree } from "../../components/visualization/json-visualization/json-tree";
import { MonacoEditor } from "../../components/code-editor";
import { tokenize } from "@designto/token";
import * as react from "@designto/react";
```


to-flutter.tsx
```
import React, { useState } from "react";
import styled from "@emotion/styled";
import { MainImageRepository } from "@design-sdk/core/assets-repository";
import { ImageRepositories } from "@design-sdk/figma/asset-repository";
import { figmacomp } from "../../components";
import { ReflectSceneNode } from "@design-sdk/core/nodes";
import { DefaultEditorWorkspaceLayout } from "../../layout/default-editor-workspace-layout";
import { LayerHierarchy } from "../../components/editor-hierarchy";
import { PreviewAndRunPanel } from "../../components/preview-and-run";
import { FigmaTargetNodeConfig } from "@design-sdk/core/utils/figma-api-utils";
import {
  WorkspaceContentPanel,
  WorkspaceContentPanelGridLayout,
} from "../../layout/panel";
// import * as flutter from "@designto/flutter";
import { MonacoEditor } from "../../components/code-editor";
import { composeAppWithHome } from "@bridged.xyz/flutter-builder";
```

The error causing part is `@bridged.xyz/flutter-builder`

When i remove the import of this package, it will work as expected. May be something inside it is crashing the webpack config. don't know how and why.

The package is developed by myself, so ask me anything about the package. but the monaco editor itself should be tolerable regardingless to it.


I'm taking the snapshot as a branch.

https://github.com/bridgedxyz/designto-code/tree/monaco-editor-issue-237




See - https://github.com/bridgedxyz/designto-code/blob/monaco-editor-issue-237/editor/pages/figma/to-flutter.tsx
Line 20 ~ 25
```
////
//// - https://github.com/suren-atoyan/monaco-react/issues/237
//// OTHER IMPORTS ARE DISABLED TEMPORARILY.
///// THIS LINE BELOW WILL BREAK/RUN THE MONACO EDITOR.
////
composeAppWithHome(""); /** COMMENT THIS LINE IF WANT TO MAKE IT WORK. */

```

**Comment by softmarshmallow** [comment]
TL;DR importing and using https://github.com/bridgedxyz/flutter-builder will throw that error.

**Comment by softmarshmallow** [comment]
Think this is caused by usage of decorator in the package.

The custom decorator is supported by `"@abraham/reflection": "^0.7.0"` - maybe this is crashing something. IDK

**Comment by tripodsgames** [comment]
> Think this is caused by usage of decorator in the package.
> 
> The custom decorator is supported by `"@abraham/reflection": "^0.7.0"` - maybe this is crashing something. IDK

Have you tested with version 0.8? 
I noticed some other things were fixed with the static method fix.

**Comment by suren-atoyan** [comment]
I close this due to inactivity. Feel free to re-open it.
